### PR TITLE
Support Inflation In Every Block

### LINKED ISSUE #461: Block inflation
An inflaction is generated for every block like as the coinbase.

### LINKED ISSUE #481: wrong error log message `invalid proposer transaction found`
```
siyonglee:sebak charles$ docker run --net host --rm -it --env-file=docker/SELF.env sebak node --genesis=GDIRF4UWPACXPPI4GW7CMTACTCNDIKJEHZK44RITZB4TD3YUM6CCVNGJ,GDYIHSHMDXJ4MXE35N4IMNC2X3Q3F665C5EX2JWHHCUW2PCFVXIFEE2C --log-level=info
INFO[10-04|00:59:54] genesis block created                    module=main height=1 round=0 timestamp=2018-10-04T00:59:54+0000 total txs=0 proposer=
INFO[10-04|00:59:54] Starting Sebak                           module=main caller=run.go:280
EROR[10-04|00:59:57] error occurred in BallotTransactionChecker module=noderunner node=GDIR.M6CC error="{\"code\":160,\"message\":\"invalid proposer transaction found\",\"data\":{}}"
INFO[10-04|00:59:58] NewBlock created                         module=noderunner node=GDIR.M6CC height=2 round=0 timestamp=2018-10-04T00:59:58+0000 total-txs=1 proposer=GDIRF4UWPACXPPI4GW7CMTACTCNDIKJEHZK44RITZB4TD3YUM6CCVNGJ
EROR[10-04|01:00:04] error occurred in BallotTransactionChecker module=noderunner node=GDIR.M6CC error="{\"code\":160,\"message\":\"invalid proposer transaction found\",\"data\":{}}"
INFO[10-04|01:00:04] NewBlock created                         module=noderunner node=GDIR.M6CC height=3 round=0 timestamp=2018-10-04T01:00:04+0000 total-txs=1 proposer=GDIRF4UWPACXPPI4GW7CMTACTCNDIKJEHZK44RITZB4TD3YUM6CCVNGJ
EROR[10-04|01:00:09] error occurred in BallotTransactionChecker module=noderunner node=GDIR.M6CC error="{\"code\":160,\"message\":\"invalid proposer transaction found\",\"data\":{}}"
INFO[10-04|01:00:09] NewBlock created                         module=noderunner node=GDIR.M6CC height=4 round=0 timestamp=2018-10-04T01:00:09+0000 total-txs=1 proposer=GDIRF4UWPACXPPI4GW7CMTACTCNDIKJEHZK44RITZB4TD3YUM6CCVNGJ
```

When I run self mode or multiple nodes, `invalid proposer transaction found` is occurred in every block.
https://github.com/bosnet/sebak/blob/master/lib/node/runner/node_runner.go#L501
->
https://github.com/bosnet/sebak/blob/master/lib/node/runner/checker_ballot_transaction.go#L152
In this line, checker.Ballot is empty.

### PR DESCRIPTION
**Pull request by spikeekips**
# Github Issues

resolves #461 #308 
resolves #481 

related to #398

## Background

When block is confirmed and stored, the fixed amount will be raised and sent to the common account.

## Solution

Like the process of transaction fee for common account, block inflation also will be processed in same way. The major changes are;

* 2 constants defined: 
    * `InflationRatio float64 = 0.0000001`
    * `BlockHeightEndOfInflation uint64 = 36000000`

* `common.InflationRatio` will keep the inflation ratio.
* `common.BlockHeightEndOfInflation`; if block height is over it, the inflation will be stopped.

* The inflation amount will be calculated in `common.CalculateInflation()`;
```
func CalculateInflation(initialBalance Amount) (a Amount, err error) {
	if initialBalance > MaximumBalance {
		err = errors.ErrorMaximumBalanceReached
		return
	}

	a = Amount(uint64(math.Round(float64(initialBalance) * InflationRatio)))
	return
}
```
* By default, `InflationRatio` will be used. It's value is `0.0000001`. This will increase `50BOS` in every block(*this amount is concrete value of our policy*).
* New operation type was added, `OperationInflation` and it's new body, `OperationBodyInflation`
* `OperationInflation` will be included in `ProposerTransaction` like `OperationCollectTxFee`.

### CODE REVIEW

File: lib/common/constant.go [hunk 1]
```diff
@@ -29,7 +29,14 @@ var (
...
 	// MaxOperationsInTransaction limits the maximum number of `Operation`s in
 	// one `Transaction`.
 	MaxOperationsInTransaction int = 1000
+
+	// DefaultInflationRatio is the default inflation ratio. If the decimal
+	// points is over 17, the inflation amount will be 0, considering with
+	// `MaximumBalance`. The current value, `0.00000000000001` will increase
+	// `50BOS` in every block(current genesis balance is `5000000000000000`).
+	DefaultInflationRatio float64 = 0.00000000000001
 )
```
Comment (soonkuk) [inline lib/common/constant.go]: I think DefaultInflationRatio should be 0.0000001.
Because current unit is GON.

Comment (spikeekips) [reply lib/common/constant.go]: @soonkuk Yes, you right. I was confused the unit. Thanks.

**Comment by zzim2x** [comment]
There is no termination condition. As far as I know 50BOS are issued per block for first 35 million blocks.

**Comment by soonkuk** [comment]
@spikeekips @zzim2x I asked to PF team about this.
They said block inflation should be generated as described in whitepaper 1.0.
The paper says 1,800,000,000 BOS should be issued through block inflation.
Every block makes 50BOS.
1800000000 / 50 = 36000000
So block infation duration is 36,000,000 blockheights.
After 36,000,000 block. No block inflation.

**Comment by kfangw** [comment]
> @spikeekips @zzim2x I asked to PF team about this. They said block inflation should be generated as described in whitepaper 1.0. The paper says `1,800,000,000 BOS` should be issued through block inflation.
> 
> Every block makes `50BOS`, `1800000000 / 50 = 36000000`.
> So block inflation duration is `36,000,000` blockheights.
> After `36,000,000` blocks, No block inflation.

After 5.7 years later (36,000,000 * 5 /60 / 60 / 24 / 365), is the inflation is stoped?

**Comment by soonkuk** [comment]
> > @spikeekips @zzim2x I asked to PF team about this. They said block inflation should be generated as described in whitepaper 1.0. The paper says `1,800,000,000 BOS` should be issued through block inflation.
> > Every block makes `50BOS`, `1800000000 / 50 = 36000000`.
> > So block inflation duration is `36,000,000` blockheights.
> > After `36,000,000` blocks, No block inflation.
> 
> After 5.7 years later (36,000,000 * 5 /60 / 60 / 24 / 365), is the inflation is stoped?

Yes That's right. According to PF team's explanation, only considerable thing is that 1800000000 BOS'll be issued by 50BOS/block. They did not care how much time spent.

**Pull request merged**